File: leetcode/addStrings.java
class Solution {
  public String addStrings(String num1, String num2) {
      
      int n = num1.length();
      int m = num2.length();
      int carr = 0;
      
      StringBuffer sb = new StringBuffer();
      
      while(n > 0 || m > 0 || carr != 0) {
          int a = (n < 1)? 0 : num1.charAt(n-1) - '0';
          int b = (m < 1)? 0 : num2.charAt(m-1) - '0';
          int num = a + b + carr;
          carr = num/10;
          num %= 10;
          sb.append(num);
          n--;
          m--;
      }
      
      sb.reverse();
      
      String str = sb.toString();
      
      return str;
  }
}

File: leetcode/addStrings.js
/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */
 var addStrings = function(num1, num2) {
  let n = num1.length
  let m = num2.length
  let carr = 0

  let ans = ""
  
  while(n > 0 || m > 0 || carr > 0) {
      let a = (n < 1) ? 0 : Number(num1[n-1])
      let b = (m < 1) ? 0 : Number(num2[m-1])
      let num = a + b + carr
      carr = ~~(num/10)
      num %= 10
      ans += num
      n--;
      m--;
  }
  
  return ans.split("").reverse().join("")
};